backend/models/siteUser.model.js

```diff
@@ -67,21 +76,27 @@ userSchema.pre("save", async function (next) {
   next();
 });
 
-userSchema.pre("save", function (next) {
+SiteUserSchema.pre("save", function (next) {
   if (!this.isModified("password") || this.isNew) return next();
 
   this.passwordChangedAt = Date.now() - 1000;
   next();
 });
 
-userSchema.methods.correctPassword = async function (
+SiteUserSchema.pre(/^find/, function (next) {
+  // this points to the current query
+  this.find({ active: { $ne: false } });
+  next();
+});
+
+SiteUserSchema.methods.correctPassword = async function (
   candidatePassword,
   userPassword
 ) {
   return await bcrypt.compare(candidatePassword, userPassword);
 };
 
-userSchema.methods.changedPasswordAfter = function (JWTTimestamp) {
+SiteUserSchema.methods.changedPasswordAfter = function (JWTTimestamp) {
```

Comment:
Ez invalidálja az összes már kiadott JWT tokent ha a token kiadása óta új jelszót állított be a user?